Moda.Web/src/moda.web.reactclient/src/app/components/common/timeline/types.d.ts

```diff
@@ -1,33 +1,32 @@
-import { DataItem } from 'vis-timeline/standalone'
+import { FC } from 'react'
+import { DataItemEnhanced, TimelineOptionsTemplateFunction, DataGroup } from 'vis-timeline/standalone'
 
-export type ModaDataItem<T = any> = DataItem & {
-  itemColor?: string | undefined
+export type ModaDataItem<T = unknown> = DataItemEnhanced & {
+  itemColor?: string
   objectData?: T
 }
 
-export type ModaDataGroup<T = any> = DataGroup & {
+export type ModaDataGroup<T = unknown> = DataGroup & {
   objectData?: T
 }
 
-export interface ModaTimelineOptions {
-  maxHeight?: number | undefined
-  minHeight?: number | undefined
-  showCurrentTime?: boolean | undefined
+export type ModaTimelineOptions<T> = {
+  maxHeight?: number
+  minHeight?: number
+  showCurrentTime?: boolean
   start: Date
   end: Date
   min: Date
   max: Date
-  groupOrder?: string | undefined
-  template?: TimelineOptionsTemplateFunction | undefined
+  groupOrder?: string
+  template?: TimelineOptionsTemplateFunction<ModaDataItem<T>>
 }
 
-export interface RangeItemTemplateProps<T = any> {
-  item: ModaDataItem<T>
+export interface ItemTemplateProps<T> {
+  // TODO: Have some sort of discriminator or conditional in this type so that we may know for certain if its is item or group
```

Comment:
All of the templates seemingly have the same props so I merged them into one here. The downside is that `item` can either be `ModaDataItem` or `ModaDataGroup`. I will look at this after work tomorrow to see how I can improve it.

Reply:
The group template will be different in the long term.

Reply:
Made a separate group template